### PR TITLE
[RF-2124] Use enum instead of int for demo progress.

### PR DESCRIPTION
**Pull request by willjschmitt**
This makes it easier to follow and understand the significance of these numbers.

This does change the numbers that were being used, but these new numbers are more consistent, and this really is a nominal breaking change for folks who have used the demo site.
<pr-train-toc>

#### PR chain:
#77 ([RF-2124] Turn on check-module for whitespace tslint rule.)
👉 #66 ([RF-2124] Use enum instead of int for demo progress.) 👈 **YOU ARE HERE**
#67 ([RF-2124] Corrects naming of payment response.)
#68 ([RF-2124] Rename references from user-id to beneficiary-id)
#69 ([RF-2124] Move reset to bottom of storage service.)
#70 ([RF-2124] Stop caching plaid token.)
#71 ([RF-2124] Make the public getter storage interface exclusively observables.)
#72 ([RF-2124] Make current step entirely calculated.)
#73 ([RF-2124] Split main component into switch/case components.)

</pr-train-toc>

[RF-2124]: https://rightfoot-eng.atlassian.net/browse/RF-2124?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RF-2124]: https://rightfoot-eng.atlassian.net/browse/RF-2124?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RF-2124]: https://rightfoot-eng.atlassian.net/browse/RF-2124?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RF-2124]: https://rightfoot-eng.atlassian.net/browse/RF-2124?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RF-2124]: https://rightfoot-eng.atlassian.net/browse/RF-2124?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RF-2124]: https://rightfoot-eng.atlassian.net/browse/RF-2124?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RF-2124]: https://rightfoot-eng.atlassian.net/browse/RF-2124?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RF-2124]: https://rightfoot-eng.atlassian.net/browse/RF-2124?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RF-2124]: https://rightfoot-eng.atlassian.net/browse/RF-2124?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: client/app/payment-form.component.ts [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
+import {Payment} from 'rightfoot-node/1-3/api';
 
-import { UnitedStatesTerritories } from './us-states-and-territories';
-import { phoneNumberValidator } from './phone-number.validator';
-import { RightfootApiService } from './rightfoot-api.service';
-import { DemographicsFormValue } from './demographics-form-value.interface';
-import { StorageService } from './storage.service';
-import { PlaidService } from './plaid.service';
+import {UnitedStatesTerritories} from './us-states-and-territories';
+import {phoneNumberValidator} from './phone-number.validator';
```
Comment (pzahynailoLohika) [inline client/app/payment-form.component.ts:9]: Please, revert the changes that remove spaces within import brackets. Perhaps your IDE is not configured to preserve/add them. I see that this happens through all files.

**Review by pzahynailoLohika — CHANGES_REQUESTED** [review]

File: client/app/storage.service.ts [hunk 2]
```diff
@@ -29,7 +30,7 @@ export class StorageService {
    */
   public storedPaymentUuid$ = this.storedPaymentUuidSubject.asObservable();
 
-  private storedCurrentStepSubject: BehaviorSubject<number> = new BehaviorSubject<number>(
+  private storedCurrentStepSubject: BehaviorSubject<DemoProgress> = new BehaviorSubject<number>(
```
Comment (akuchibotla) [inline client/app/storage.service.ts:33]: Shouldn't this be
```suggestion
  private storedCurrentStepSubject: BehaviorSubject<DemoProgress> = new BehaviorSubject<DemoProgress>(
```

**Review by akuchibotla — APPROVED** [review]

File: client/app/storage.service.ts (hunk 2 above)
Comment (willjschmitt) [reply client/app/storage.service.ts:33]: Yeah there were a few more I missed, but ended up cleaning up downstream. Going to dismiss for now, since it gets fixed in a downstream change

File: client/app/payment-form.component.ts (hunk 1 above)
Comment (willjschmitt) [reply client/app/payment-form.component.ts]: Corrected! Added tslint rule to catch this in the upstream
